### PR TITLE
Virtual Garden Provider implementation

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**What this PR does / why we need it**:
This PR implements the basics of a virtual garden provider. It also shares the `242376` rule logic between providers. Par of https://github.com/gardener/diki/issues/68

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
We should also extend the new version script https://github.com/gardener/diki/blob/main/hack/gen-new-disak8sstig-version.sh but that can be done in another PR.

**Release note**:

```feature developer
Diki now has a basic implementation of a virtual garden provider.
```

### CODE REVIEW

File: pkg/provider/builder/virtualgarden.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+// VirtualGardenProviderFromConfig retuns a Provider from a [ProviderConfig].
+func VirtualGardenProviderFromConfig(conf config.ProviderConfig) (provider.Provider, error) {
+	p, err := virtualgarden.FromGenericConfig(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	setConfigDefaults(p.GardenConfig)
+	setConfigDefaults(p.RuntimeConfig)
```
Comment (AleksandarSavchev) [inline pkg/provider/builder/virtualgarden.go:26]: I think the `setConfigDefaults()` function should be moved to a new utils file. ex. `pkg/provider/builder/utils.go`

File: pkg/shared/disak8sstig/v1r11/ids.go
Comment (AleksandarSavchev) [inline pkg/shared/disak8sstig/v1r11/ids.go:1]: We should remove the [pkg/provider/gardener/ruleset/disak8sstig/v1r11/options.go](https://github.com/gardener/diki/blob/main/pkg/provider/gardener/ruleset/disak8sstig/v1r11/options.go) file and use the constants from this file. We can do this in a new PR.

File: pkg/provider/virtualgarden/provider.go
Comment (AleksandarSavchev) [inline pkg/provider/virtualgarden/provider.go:1]: You can add a unit test file as we have for the `gardener` provider: [ref](https://github.com/gardener/diki/blob/main/pkg/provider/gardener/provider_test.go)

File: example/config/virtualgarden.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    gardenKubeconfigPath: /tmp/garden.config  # path to garden cluster admin kubeconfig
+    runtimeKubeconfigPath: /tmp/runtime.config  # path to runtime cluster admin kubeconfig
+  rulesets:
+  - id: disa-kubernetes-stig
+    name: DISA Kubernetes Security Technical Implementation Guide
+    version: v1r11
+    ruleOptions: []
+output:
+  path: /tmp/test-output.json          #  optional, path to summary json report
+  minStatus: Passed
```
Comment (AleksandarSavchev) [inline example/config/virtualgarden.yaml:16]:
```suggestion
providers:                # contains information about known providers
- id: virtualgarden       # unique provider identifier
  name: "Virtual Garden"  # user friendly name of the provider
  metadata:
    foo: bar
  args:
    gardenKubeconfigPath: /tmp/garden.config    # path to garden cluster admin kubeconfig
    runtimeKubeconfigPath: /tmp/runtime.config  # path to runtime cluster admin kubeconfig
  rulesets:
  - id: disa-kubernetes-stig
    name: DISA Kubernetes Security Technical Implementation Guide
    version: v1r11
    ruleOptions: []
output:
  path: /tmp/test-output.json  #  optional, path to summary json report
  minStatus: Passed
```

**Review by AleksandarSavchev — CHANGES_REQUESTED** [review]
I think we should move the `shared` dir into `pkg/provider/shared`. WDYT?

File: pkg/provider/builder/virtualgarden.go (hunk 1 above)
Comment (dimityrmirchev) [reply pkg/provider/builder/virtualgarden.go:26]: I think that there is no need for another file right now because it is a single function

**Comment by dimityrmirchev** [comment]
> I think we should move the shared dir into pkg/provider/shared. WDYT?

I addd a `ruleset` folder in between. I think that should be fine for now

**Review by AleksandarSavchev — APPROVED** [review]
/lgtm

**Pull request merged**